src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1901,9 +1922,31 @@ private void doGenKeyPair(String alias, String dname, String keyAlgName,
             throw new Exception(form.format(source));
         }
 
-        CertAndKeyGen keypair =
-                new CertAndKeyGen(keyAlgName, sigAlgName, providerName);
+        CertAndKeyGen keypair;
+        if (signerAlias != null) {
+            signerFlag = true;
```

Comment:
Is the `signerFlag` really useful? We can just check `signerAlias != null`.

Reply:
Sure, use `signerAlias` instead.